x-pack/plugins/enterprise_search/server/lib/enterprise_search_config.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { readFileSync } from 'fs';
+
+import http from 'http';
+import https from 'https';
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+// -----------------------------------------------------------------------------
+export const configSchema = schema.object({
+  host: schema.maybe(schema.string()),
+  enabled: schema.boolean({ defaultValue: true }),
+  accessCheckTimeout: schema.number({ defaultValue: 5000 }),
+  accessCheckTimeoutWarning: schema.number({ defaultValue: 300 }),
+  ssl: schema.object({
+    certificateAuthorities: schema.maybe(
+      schema.oneOf([schema.arrayOf(schema.string()), schema.string()])
+    ),
+    rejectUnauthorized: schema.boolean({ defaultValue: false }),
+  }),
```

Comment:
Another quick note - if we're changing our config yml options, I _believe_ that's considered an API change by Kibana and we need to document it. I'm a terrible person and I haven't actually looked into what goes into that but I think it's something that gets auto generated into https://github.com/elastic/kibana/blob/master/api_docs/enterprise_search.json 😬 I'll probably have to ask in the Kibana core channel